flutter_mobx_forms/lib/src/form_field.dart

```diff
@@ -0,0 +1,92 @@
+import 'package:mobx/mobx.dart';
+
+part 'form_field.g.dart';
+
+typedef SyncFieldValidator<T> = Iterable<String> Function(T value);
+typedef AsyncFieldValidator<T> = Stream<String> Function(T value);
+
+enum ValidationPolicy { onChange, always, manual }
+
+class FormField<T> = _FormField<T> with _$FormField;
+
+abstract class _FormField<T> with Store {
+  _FormField(
+      {this.name,
+      this.label,
+      this.value,
+      this.validator,
+      this.asyncValidator,
+      this.validationPolicy = ValidationPolicy.manual}) {
+    // ignore: prefer_asserts_in_initializer_lists
+    assert(validator != null && asyncValidator != null,
```

Comment:
can't it be:

```dart
  _FormField({this.name,
      this.label,
      this.value,
      this.validator,
      this.asyncValidator,
      this.validationPolicy = ValidationPolicy.manual,
  }): assert(validator != null && asyncValidator != null, 'Only one of validator or asyncValidator can be specified');
```

?

Reply:
Definitely could, just felt like a really loaded constructor line :-). If that's the idiomatic approach, I'm happy to change. .... actually changed